app/objects/c_ability.py

```diff
@@ -140,25 +89,23 @@ async def which_plugin(self):
                 return plugin
         return None
 
-    def replace_cleanup(self, encoded_cmd, payload):
-        decoded_cmd = b64decode(encoded_cmd).decode('utf-8', errors='ignore').replace('\n', '')
-        decoded_cmd = decoded_cmd.replace(self.RESERVED['payload'], payload)
-        return decoded_cmd
+    def get_executor(self, platform, name):
+        return next(iter([e for e in self.executors if e.platform == platform and e.name == name]), None)
+
+    def get_executors(self, platform, names):
+        executors = []
+        for name in names:
+            matched_executor = self.get_executor(platform, name)
+            if matched_executor and matched_executor not in executors:
```

Comment:
The `in` for a list is gonna be O(n) on each iteration. Should `executors` be a `set`? Do we care if there are duplicates or ordering?

Reply:
There shouldn't be duplicates if `names` are unique, but I don't want to rely on that. We don't care about order here, that is handled by agents that are selecting the "preferred" executor from this list.

Changed `executors` to a `set`.